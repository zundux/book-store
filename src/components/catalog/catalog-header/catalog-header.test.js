import React from "react";
import CatalogHeader from "./catalog-header";

describe("Catalog grid header", () => {

  test("Catalog header has not only the text \"Catalog\" when catalog is empty", () => {
    const component = shallow(
      <CatalogHeader productsInCart={[]}/>
    );
    matchSnapshot(component);
  });

  test("Header has the word \"product\" when cart has only 1 product", () => {
    const component = shallow(
      <CatalogHeader productsInCart={[
        {
          id: 1,
          amount: 1
        }
      ]}/>
    );
    matchSnapshot(component);
  });

  test("Header has the word \"products\" when cart has a couple of products", () => {
    const component = shallow(
      <CatalogHeader productsInCart={[
        {
          id: 1,
          amount: 1
        },
        {
          id: 2,
          amount: 1
        }
      ]}/>
    );
    matchSnapshot(component);
  });

  test("Header has the word \"products\" when cart contain 2 copies of the product", () => {
    const component = shallow(
      <CatalogHeader productsInCart={[
        {
          id: 1,
          amount: 2
        }
      ]}/>
    );
    matchSnapshot(component);
  });

});

