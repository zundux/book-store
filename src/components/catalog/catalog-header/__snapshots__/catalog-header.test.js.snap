// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Catalog grid header Catalog header has not only the text "Catalog" when catalog is empty 1`] = `
<div>
  <h1>
    Catalog
  </h1>
</div>
`;

exports[`Catalog grid header Header has the word "product" when cart has only 1 product 1`] = `
<div>
  <h1>
    Catalog
  </h1>
  <div>
    <Link
      replace={false}
      to="/cart"
    >
      1
       product
       in cart
    </Link>
  </div>
</div>
`;

exports[`Catalog grid header Header has the word "products" when cart contain 2 copies of the product 1`] = `
<div>
  <h1>
    Catalog
  </h1>
  <div>
    <Link
      replace={false}
      to="/cart"
    >
      2
       product
      s
       in cart
    </Link>
  </div>
</div>
`;

exports[`Catalog grid header Header has the word "products" when cart has a couple of products 1`] = `
<div>
  <h1>
    Catalog
  </h1>
  <div>
    <Link
      replace={false}
      to="/cart"
    >
      2
       product
      s
       in cart
    </Link>
  </div>
</div>
`;
